### PR TITLE
CARP Team Six: Nanotrasen's Finest*

### PR DESCRIPTION
**Pull request by Paxilmaniac**
## About The Pull Request
You didn't ask for their help, but they were sent regardless.
image
(The armor is all from tgmc, so credit to all the different people the armor bits are from)

Adds an ert in the form of a squad of marines, featuring the commander, the heavy, the medic, and the engineer, all with their own sets of armor and equipment.

The team comes with wt550 autorifles, and a shotgun if you get engineer (yeehaw), and are absolutely not spaceproof.

image
(Marines hard at work dying and shooting at things)

## Why It's Good For The Game
Sometimes you'll want something that has more gear than interns, but less than what the other erts have to offer, or you just like watching swarms of tacticool marine roleplayers shoot shit with guns.

## Changelog
:cl:
add: New ERT type, marines, for whatever you might want them for.
add: Tacticool armor from tgmc to compliment.
/:cl:

### CODE REVIEW

File: code/datums/ert.dm [hunk 1]
```diff
@@ -41,6 +41,13 @@
 	mission = "Leave no witnesses."
 	polldesc = "an elite Nanotrasen Strike Team"
 
+/datum/ert/marine
+	leader_role = /datum/antagonist/ert/marine
+	roles = list(/datum/antagonist/ert/marine/security, /datum/antagonist/ert/marine/engineer, /datum/antagonist/ert/marine/medic)
+	rename_team = "Marine Squad"
+	polldesc = "an elite Nanotrasen Strike Team"
```
Comment (Fikou) [inline code/datums/ert.dm:48]: this probably shouldnt be the same message as the death squad

File: code/modules/antagonists/ert/ert.dm [hunk 2]
```diff
@@ -351,3 +351,23 @@
...
+
+/datum/antagonist/ert/marine
+	name = "Marine Commander"
+	outfit = /datum/outfit/centcom/ert/marine
+	role = "Commander"
+
+/datum/antagonist/ert/marine/security
+	name = "Marine Heavy"
+	outfit = /datum/outfit/centcom/ert/marine/security
+	role = "Security Officer"
```
Comment (Fikou) [inline code/modules/antagonists/ert/ert.dm:363]: he aint a security officer doe

Comment (Paxilmaniac) [reply code/modules/antagonists/ert/ert.dm:363]: you see this is copied from the ert sec officer and i didn't read what it actually does 😎

File: code/datums/ert.dm (hunk 1 above)
Comment (Paxilmaniac) [reply code/datums/ert.dm]: it should be better now

**Comment by AnCopper** [comment]
There are TerraGov Soldiers outside of my Office.

File: code/modules/clothing/outfits/ert.dm [hunk 3]
```diff
@@ -537,3 +537,111 @@
...
+	mask = /obj/item/clothing/mask/cigarette/robustgold
+	head = /obj/item/clothing/head/helmet/marine
+	l_hand = /obj/item/flashlight/seclite
+
+/datum/outfit/centcom/ert/marine/post_equip(mob/living/carbon/human/H, visualsOnly = FALSE)
+	..()
+
+	if(visualsOnly)
+		return
+	var/obj/item/radio/R = H.ears
```
Comment (tralezab) [inline code/modules/clothing/outfits/ert.dm:562]: please use better variable names

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/modules/clothing/outfits/ert.dm [hunk 4]
```diff
@@ -537,3 +537,111 @@
...
+	belt = /obj/item/storage/belt/military/assault/full
+	glasses = /obj/item/clothing/glasses/hud/security/sunglasses/eyepatch
+	l_pocket = /obj/item/kitchen/knife/combat
+	r_pocket = /obj/item/lighter
+	uniform = /obj/item/clothing/under/syndicate/camo
+	mask = /obj/item/clothing/mask/cigarette/robustgold
+	head = /obj/item/clothing/head/helmet/marine
+	l_hand = /obj/item/flashlight/seclite
+
+/datum/outfit/centcom/ert/marine/post_equip(mob/living/carbon/human/H, visualsOnly = FALSE)
```
Comment (tralezab) [inline code/modules/clothing/outfits/ert.dm:557]: please use better variable names

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/modules/clothing/outfits/ert.dm [hunk 5]
```diff
@@ -537,3 +537,111 @@
...
+	suit_store = /obj/item/gun/ballistic/automatic/wt550
+	back = /obj/item/shield/riot
+	belt = /obj/item/storage/belt/military/assault/full
+	glasses = /obj/item/clothing/glasses/hud/security/sunglasses
+	r_pocket = /obj/item/grenade/smokebomb
+	mask = /obj/item/clothing/mask/gas/sechailer
+	head = /obj/item/clothing/head/helmet/marine/security
+	l_hand = /obj/item/flashlight/seclite
+
+/datum/outfit/centcom/ert/marine/security/post_equip(mob/living/carbon/human/H, visualsOnly = FALSE)
```
Comment (tralezab) [inline code/modules/clothing/outfits/ert.dm:580]: please use better variable names

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/modules/clothing/outfits/ert.dm [hunk 6]
```diff
@@ -537,3 +537,111 @@
...
+	..()
+
+	if(visualsOnly)
+		return
+
+	var/obj/item/radio/R = H.ears
```
Comment (tralezab) [inline code/modules/clothing/outfits/ert.dm:586]: please use better variable names

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/modules/clothing/outfits/ert.dm [hunk 7]
```diff
@@ -537,3 +537,111 @@
...
+	backpack_contents = list(
+		/obj/item/reagent_containers/hypospray/combat = 1,
+		/obj/item/storage/firstaid/regular = 1,
+		/obj/item/storage/firstaid/advanced = 1
+)
+	belt = /obj/item/storage/belt/medical/paramedic
+	glasses = /obj/item/clothing/glasses/hud/health/sunglasses
+	l_hand = /obj/item/flashlight/seclite
+
+/datum/outfit/centcom/ert/marine/medic/post_equip(mob/living/carbon/human/H, visualsOnly = FALSE)
```
Comment (tralezab) [inline code/modules/clothing/outfits/ert.dm:609]: please use better variable names

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/modules/clothing/outfits/ert.dm [hunk 8]
```diff
@@ -537,3 +537,111 @@
...
+	..()
+
+	if(visualsOnly)
+		return
+
+	var/obj/item/radio/R = H.ears
```
Comment (tralezab) [inline code/modules/clothing/outfits/ert.dm:615]: please use better variable names

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/modules/clothing/outfits/ert.dm [hunk 9]
```diff
@@ -537,3 +537,111 @@
...
+		/obj/item/pipe_dispenser = 1,
+		/obj/item/storage/box/lethalshot = 1,
+		/obj/item/grenade/c4 = 3
+)
+	belt = /obj/item/storage/belt/utility/full/powertools
+	glasses =  /obj/item/clothing/glasses/night
+	r_pocket = /obj/item/rcd_ammo/large
+	l_hand = /obj/item/flashlight/seclite
+
+/datum/outfit/centcom/ert/marine/engineer/post_equip(mob/living/carbon/human/H, visualsOnly = FALSE)
```
Comment (tralezab) [inline code/modules/clothing/outfits/ert.dm:639]: please use better variable names

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/modules/clothing/outfits/ert.dm [hunk 10]
```diff
@@ -537,3 +537,111 @@
...
+	..()
+
+	if(visualsOnly)
+		return
+
+	var/obj/item/radio/R = H.ears
```
Comment (tralezab) [inline code/modules/clothing/outfits/ert.dm:645]: please use better variable names

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/game/objects/items/storage/belt.dm [hunk 11]
```diff
@@ -484,6 +484,12 @@
 	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
 	STR.max_items = 6
 
+/obj/item/storage/belt/military/assault/full/PopulateContents()
+	var/static/items_inside = list(
```
Comment (tralezab) [inline code/game/objects/items/storage/belt.dm:488]: ...static? i'd rather this be `var/list/items_inside`

File: code/game/objects/items/storage/belt.dm [hunk 12]
```diff
@@ -484,6 +484,12 @@
...
+		/obj/item/ammo_box/magazine/wt550m9/wtap = 2,
+		/obj/item/ammo_box/magazine/wt550m9 = 4)
```
Comment (tralezab) [inline code/game/objects/items/storage/belt.dm:490]: indent the ) to a new line, lists should be formatted like this:

```dm
var/list = list(
    thing,
    thing2
)
```

**Review by tralezab — COMMENTED** [review]
here's a review to improve the code

File: code/game/objects/items/storage/belt.dm [hunk 13]
```diff
@@ -484,6 +484,13 @@
 	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
 	STR.max_items = 6
 
+/obj/item/storage/belt/military/assault/full/PopulateContents()
+	var/list/items_inside = list(
+		/obj/item/ammo_box/magazine/wt550m9/wtap = 2,
+		/obj/item/ammo_box/magazine/wt550m9 = 4
+	)
+	generate_items_inside(items_inside,src)
```
Comment (Mothblocks) [inline code/game/objects/items/storage/belt.dm:492]: Why create the variable if you're just going to use it once?

File: code/game/objects/items/storage/belt.dm [hunk 14]
```diff
@@ -484,6 +484,13 @@
 	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
 	STR.max_items = 6
 
+/obj/item/storage/belt/military/assault/full/PopulateContents()
+	var/list/items_inside = list(
+		/obj/item/ammo_box/magazine/wt550m9/wtap = 2,
+		/obj/item/ammo_box/magazine/wt550m9 = 4
```
Comment (Mothblocks) [inline code/game/objects/items/storage/belt.dm:490]: Use trailing commas.

File: code/game/objects/items/storage/holsters.dm [hunk 15]
```diff
@@ -61,6 +61,20 @@
...
+	icon_state = "syndicate_holster"
+	inhand_icon_state = "syndicate_holster"
+	worn_icon_state = "syndicate_holster"
+
+/obj/item/storage/belt/holster/detective/full/ert/PopulateContents()
+	var/list/items_inside = list(
+		/obj/item/gun/ballistic/automatic/pistol/m1911 = 1,
+		/obj/item/ammo_box/magazine/m45 = 2
+	)
+	generate_items_inside(items_inside,src)
```
Comment (Mothblocks) [inline code/game/objects/items/storage/holsters.dm:76]: Same here.

File: code/modules/clothing/outfits/ert.dm [hunk 16]
```diff
@@ -537,3 +537,107 @@
...
+	suit = /obj/item/clothing/suit/armor/vest/marine/engineer
+	suit_store = /obj/item/gun/ballistic/shotgun/lethal
+	mask = /obj/item/clothing/mask/cigarette/robustgold
+	head = /obj/item/clothing/head/helmet/marine/engineer
+	back = /obj/item/storage/backpack/ert/engineer
+	backpack_contents = list(
+		/obj/item/construction/rcd/loaded/upgraded = 1,
+		/obj/item/pipe_dispenser = 1,
+		/obj/item/storage/box/lethalshot = 1,
+		/obj/item/grenade/c4 = 3
```
Comment (Mothblocks) [inline code/modules/clothing/outfits/ert.dm:629]:
```suggestion
		/obj/item/grenade/c4 = 3,
```

Helps prevent ugly blames, looks consistent.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Pull request merged**